inc/stm32_compat.h

```diff
@@ -0,0 +1,59 @@
+#ifndef _STM32_COMPAT_H_
+#define _STM32_COMPAT_H_
+
+#ifndef PLATFORMIO
+
+#include <stm32.h>
+
+#else // PLATFORMIO
```

Comment:
With this file present it doesn't make sense to me to retain the dependency on `stm32.h`. I.e. the conditional here creates unnecessary complexity.

1. Either use `stm32_compat.h` in all cases

2. Or retain dependency on `stm32.h` in all cases and fetch the file dependency under PIO (https://docs.platformio.org/en/latest/manifests/library-json/fields/dependencies.html)

Reply:
regarding the second option, turning the stm32h repository with it's current form into a dependency does not make sense to me. 

problem is since it uses submodules for all variants of st family, adding it to a project would re-download the libraries that are already handled by PIO, which is not bandwidth- and space-efficient.

Reply:
I think it would be viable to have that file in a repo without the submodules (@dmitrystu ?)